app/assets/js/authmanager.js

```diff
@@ -77,23 +131,31 @@ exports.removeAccount = async function(uuid){
 exports.validateSelected = async function(){
     const current = ConfigManager.getSelectedAccount()
     const isValid = await Mojang.validate(current.accessToken, ConfigManager.getClientToken())
-    if(!isValid){
-        try {
-            const session = await Mojang.refresh(current.accessToken, ConfigManager.getClientToken())
-            ConfigManager.updateAuthAccount(current.uuid, session.accessToken)
-            ConfigManager.save()
-        } catch(err) {
-            logger.debug('Error while validating selected profile:', err)
-            if(err && err.error === 'ForbiddenOperationException'){
-                // What do we do?
-            }
-            logger.log('Account access token is invalid.')
-            return false
-        }
-        loggerSuccess.log('Account access token validated.')
+
+    if (isValid) {
         return true
+    }
+
+    if (ConfigManager.getSelectedAccount() === 'microsoft') {
+        const validate = await validateSelectedMicrosoft()
+        return validate
     } else {
-        loggerSuccess.log('Account access token validated.')
-        return true
+        const validate = await validateSelectedMojang()
+        return validate
     }
-}
\ No newline at end of file
+}
+
+exports.addMSAccount = async authCode => {
+    const accessToken = await Microsoft.getAccessToken(authCode)
+    const MCAccessToken = await Microsoft.authMinecraft(accessToken.access_token)
+    const minecraftBuyed = await Microsoft.checkMCStore(MCAccessToken.access_token)
+    if (!minecraftBuyed)
+        throw {
+            message: 'You didn\'t buy Minecraft! Please use another Microsoft account or buy Minecraft.'
+        }
+    const MCProfile = await Microsoft.getMCProfile(MCAccessToken.access_token)
```

Comment:
To support Game Pass, the purchase check will need to be skipped since it won't detect the game:
```suggestion
    let MCProfile

    try {
        MCProfile = await Microsoft.getMCProfile(MCAccessToken.access_token)
    } catch {
        throw {
            message: 'You didn\'t buy Minecraft! Please use another Microsoft account or buy Minecraft.'
        }
    }
```